CMakeLists.txt

```diff
@@ -270,8 +270,9 @@ if (NOT ("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC"))
 		add_compile_options("-D_REENTRANT")
 	endif()
 
-	if (DISABLE_BSYMBOLIC STREQUAL "OFF")
-		set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKED_FLAGS} -Wl,-Bsymbolic-functions")
+	# OSX Mach-O doesn't support linking with '-Bsymbolic-functions'.
+	if (DISABLE_BSYMBOLIC STREQUAL "OFF" AND NOT APPLE)
```

Comment:
I'd really rather not hardcode a particular vendor like this.  If we're trying to use a linker flag it would be better to actually check whether it works (#569).

Reply:
Fixed.